daemon/images/image_exporter.go

```diff
@@ -13,15 +13,30 @@ import (
 // stream. All images with the given tag and all versions containing
 // the same tag are exported. names is the set of tags to export, and
 // outStream is the writer which the images are written to.
-func (i *ImageService) ExportImage(ctx context.Context, names []string, platform *ocispec.Platform, outStream io.Writer) error {
+func (i *ImageService) ExportImage(ctx context.Context, names []string, platformList []*ocispec.Platform, outStream io.Writer) error {
+	var platform *ocispec.Platform
+
+	// The docker image store only supports single platform images. If multiple platforms are given,
+	// use the last one (for compatibility with previous behavior).
+	if len(platformList) > 0 {
+		platform = platformList[len(platformList)-1]
+	}
```

Comment:
Considering if we should make this check for the overlap between the given list of platforms and the image that's specified.

Basically thinking;

- `docker pull --platform linux/s390x alpine`
- I now have `alpine` on my machine, but only as `linux/s390x`
- `docker save --platform linux/s390x,linux/amd64 alpine` (save `alpine`,  but only the `s390x` and `amd64` variants, skip other variants
- ^^ we only have `s390x`, and "skip" `amd64` (which ... isn't present, so nothing to skip)

I guess that depends a bit on how we describe / consider the purpose of `--platform` in this case;

- Is it a requirement? (export if present, fail if not)
- Is it a filter? (export the given platforms, and skip other ones)